adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -428,17 +438,24 @@ def connect(
                 self.logger.warning(f"Socket error when connecting: {e}")
                 backoff = False
             except MMQTTException as e:
-                last_exception = e
+                self._close_socket()
```

Comment:
After helping a bunch of people over the last week, I realized users were getting `Repeated connect failures`, and the second+ all errored that the `socket was already open`. This fixes that so it's closed on the next attempt